resources/kernel/make_busybox.sh

```diff
@@ -0,0 +1,213 @@
+BUSYBOX_ARCHIVE="$BUSYBOX.tar.bz2"
+BUSYBOX_URL="https://busybox.net/downloads/$BUSYBOX_ARCHIVE"
+
+# Reset index for cmdline arguments for the following `getopts`.
+OPTIND=1
+# Flag for optionally building a guest that halts.
+HALT=
+# Number of CPUs to use during the kernel build process.
+MAKEPROCS=1
+# Flag for optionally cleaning the workdir and recompiling the kernel.
+CLEAN=
+
+# Supported arguments:
+# * `-h`: build a guest that halts.
+# * `-j`: compile with `make -j` (on all available CPUs).
+#         TODO: pass the value for `-j`.
+while getopts "chj:" opt; do
+    case "$opt" in
+    c)  CLEAN=1
+        ;;
+    h)  HALT=1
+        ;;
+    j)  MAKEPROCS=$OPTARG
+        ;;
+    *)  ;; # Ignore other flags for now.
+    esac
+done
+shift $((OPTIND-1))
+
+cleanup() {
+    if [ -n "$CLEAN" ]; then
+        echo "Cleaning $WORKDIR..."
+        rm -rf "$WORKDIR"
+    fi
+}
+
+extract_kernel_srcs() {
+    echo "Starting kernel build."
+    # Download kernel sources.
+    echo "Downloading kernel..."
+    curl "$KERNEL_URL" > "$KERNEL_ARCHIVE"
+    echo "Extracting kernel sources..."
+    tar xf "$KERNEL_ARCHIVE"
+}
+
+make_kernel_config() {
+    # Copy base kernel config.
+    # Add any custom config options, if necessary (currently N/A).
+    kernel_dir="$1"
+    echo "Copying kernel config..."
+    cp "$TEST_RESOURCE_DIR/microvm-kernel-initramfs-hello-x86_64.config" "$kernel_dir/.config"
+}
+
+make_busybox() {
+    kernel_dir="$1"
+    nprocs="$2"
+    # Move to the directory with the kernel sources.
+    pushd "$kernel_dir" &>/dev/null
+    # Prepare busybox.
+    echo "Downloading busybox..."
+    mkdir -p busybox_rootfs
+    curl "$BUSYBOX_URL" > "$BUSYBOX_ARCHIVE"
+    echo "Extracting busybox..."
+    tar xf "$BUSYBOX_ARCHIVE"
+    pushd "$BUSYBOX" &>/dev/null
+    # Build statically linked busybox.
+    cp "$TEST_RESOURCE_DIR/busybox_static_config" .config
+    echo "Building busybox..."
+    make -j "$nprocs"
+    # Package all artefacts somewhere else.
+    echo "Packaging busybox..."
+    make CONFIG_PREFIX=../busybox_rootfs install
+    # Back to kernel dir.
+    popd &>/dev/null
+    # Back to wherever we were before.
+    popd &>/dev/null
+}
+
+make_init() {
+    halt="$1"
+    # Make an init script.
+    cd ..
+    echo "Creating init script..."
+    cat >init <<EOF
+#!/bin/sh
+mount -t proc none /proc
+mount -t sysfs none /sys
+/bin/echo "                                                   "
+/bin/echo "                 _                                 "
+/bin/echo "  _ __ _   _ ___| |_    __   ___ __ ___  _ __ ___  "
+/bin/echo " | '__| | | / __| __|___\ \ / / '_ \ _ \| '_ \ _ \ "
+/bin/echo " | |  | |_| \__ \ ||_____\ V /| | | | | | | | | | |"
+/bin/echo " |_|   \__,_|___/\__|     \_/ |_| |_| |_|_| |_| |_|"
+/bin/echo "                                                   "
+/bin/echo "                                                   "
+/bin/echo "Hello, world, from the rust-vmm reference VMM!"
+EOF
```

Comment:
```suggestion
mount -t sysfs none /sys
mknod -m 666 /dev/ttyS0 c 4 64
/bin/echo "                                                   "
/bin/echo "                 _                                 "
/bin/echo "  _ __ _   _ ___| |_    __   ___ __ ___  _ __ ___  "
/bin/echo " | '__| | | / __| __|___\ \ / / '_ \ _ \| '_ \ _ \ "
/bin/echo " | |  | |_| \__ \ ||_____\ V /| | | | | | | | | | |"
/bin/echo " |_|   \__,_|___/\__|     \_/ |_| |_| |_|_| |_| |_|"
/bin/echo "                                                   "
/bin/echo "                                                   "
/bin/echo "Hello, world, from the rust-vmm reference VMM!"
setsid  cttyhack sh
EOF
```

Using this get around, `/bin/sh: can't access tty; job control turned off` won't be printed anymore after the `Hello, world, ...`: https://stackoverflow.com/questions/36529881/qemu-bin-sh-cant-access-tty-job-control-turned-off. The other answer from here might be nicer, but I didn't manage to make it work yet.